ecs-agent/api/ecs/client/ecs_client.go

```diff
@@ -752,6 +766,17 @@ func (client *ecsClient) UpdateContainerInstancesState(instanceARN string, statu
 	return err
 }
 
+func formatManagedAgents(managedAgents []*ecsmodel.ManagedAgentStateChange) []*ecsmodel.ManagedAgentStateChange {
+	var result []*ecsmodel.ManagedAgentStateChange
+	for _, m := range managedAgents {
+		if m.Reason != nil {
+			m.Reason = aws.String(trimString(aws.StringValue(m.Reason), ecsMaxContainerReasonLength))
```

Comment:
What happens when reason length exceeds the maximum? Does the backend throw a validation error or something like that?

Reply:
> or reference, we already have similar trimming logic in place for other "reason" fields such as container reason

Is there any scope of reusing the existing function? If not, can we add a test for this function?

Reply:
> What happens when reason length exceeds the maximum? Does the backend throw a validation error or something like that?

Correct, error will occur due to length of reason being too large.

> Is there any scope of reusing the existing function? If not, can we add a test for this function?

Good call out, I am thinking it would be better to define a function to trim string values associated with string pointers (i.e., one that does something like `aws.String(trimString(aws.StringValue(INSERT_STRING_POINTER_HERE), INSERT_MAX_LENGTH_HERE))`), and then just reuse that function where we need to. I will do that and include testing of said function.

Reply:
Yep that will be great!